test/support/utils.h

```diff
@@ -539,8 +539,7 @@ class Wrapper
     }
     Wrapper(Wrapper&& input)
     {
-        my_field = input.my_field;
-        input.my_field = nullptr;
```

Comment:
I'm not sure that it (checking NULL) is  not used in the tests for some checks for a moved instance.
it should be double checked.

Reply:
There are no changes in run-time:
https://en.cppreference.com/w/cpp/memory/shared_ptr/shared_ptr
10): ```shared_ptr( shared_ptr&& r ) noexcept;```

"10) Move-constructs a shared_ptr from r. After the construction, *this contains a copy of the previous state of r, r is empty and its stored pointer is null. The template overload doesn't participate in overload resolution if Y* is not implicitly convertible to (until C++17)compatible with (since C++17) T*."

Reply:
Also checked in debugger under MSVC compiler.